crates/compiler/parse/src/header.rs

```diff
@@ -168,42 +189,39 @@ pub enum To<'a> {
 
 #[derive(Clone, Debug, PartialEq)]
 pub struct AppHeader<'a> {
+    pub before_name: &'a [CommentOrNewline<'a>],
     pub name: Loc<StrLiteral<'a>>,
-    pub packages: Collection<'a, Loc<Spaced<'a, PackageEntry<'a>>>>,
-    pub imports: Collection<'a, Loc<Spaced<'a, ImportsEntry<'a>>>>,
-    pub provides: Collection<'a, Loc<Spaced<'a, ExposedName<'a>>>>,
-    pub provides_types: Option<Collection<'a, Loc<Spaced<'a, UppercaseIdent<'a>>>>>,
-    pub to: Loc<To<'a>>,
 
-    // Potential comments and newlines - these will typically all be empty.
-    pub before_header: &'a [CommentOrNewline<'a>],
-    pub after_app_keyword: &'a [CommentOrNewline<'a>],
-    pub before_packages: &'a [CommentOrNewline<'a>],
-    pub after_packages: &'a [CommentOrNewline<'a>],
-    pub before_imports: &'a [CommentOrNewline<'a>],
-    pub after_imports: &'a [CommentOrNewline<'a>],
-    pub before_provides: &'a [CommentOrNewline<'a>],
-    pub after_provides: &'a [CommentOrNewline<'a>],
-    pub before_to: &'a [CommentOrNewline<'a>],
-    pub after_to: &'a [CommentOrNewline<'a>],
+    pub packages:
+        Option<KeywordItem<'a, PackagesKeyword, Collection<'a, Loc<Spaced<'a, PackageEntry<'a>>>>>>,
```

Comment:
I am reminded of [this tweet](https://twitter.com/rtfeldman/status/1569474910661279744) by @rtfeldman

Reply:
besides that, this is a nice change and should not increase the size of the type because generally the element types all have an alignment of 8 anyway.

Reply:
> I am reminded of [this tweet](https://twitter.com/rtfeldman/status/1569474910661279744)

Lol!

> should not increase the size of the type

I actually wouldn't be surprised if it did, at least a bit - particularly with the `Option` bit. Or maybe the compiler finds a place to squish that bit, in one of the nested pointers? 🤔

Anyway, even it it did increase in size a bit I think this is probably fine, since we'll only have one of these top-level types - not a bunch like we have for Exprs.